### ISSUE TITLE
Not printing over VNC

### ISSUE THREAD

**Issue by LKGUK** [PDConSec/vsc-print#260]
Hi, not sure if you are aware of this....

in Visual Studio Code, If I use the original (now deprecated for yours) 'Printcode' extension, I can print to my from my Pi over VNC to my printer connected to my PC.

Using your 'Print' extension goes through exactly the same process ending up with the print preview of the code in the Chromium print dialogue on Pi (Buster), but then sending to the same printer the print never gets printed!

Background info that may indicate something useful:
The print dialogues in Chromium on Pi Buster looks identical between 'Printcode' and 'Print' apart from the fact that in yours the 'Print Orientation option is available' and in the 'Printcode' version the option initially appears but then flashes off immediately so the option is no longer available!

In addition, printing from Notepad, Printer test page and Chromium all print fine to the same printer over VNC so I am thinking maybe this is something to do with the way the 'Print' extension invokes the print dialogue in Chromium on Pi Buster, it seems to be the only difference in all print scenarios.

I would much prefer to use your extension because it seems far superior in respect of; it prints filename as print header, It can print in landscape mode and it adds the 'print' option in VS code when code is selected.

**Comment by LKGUK** [comment]
I don't believe it!, after positing this, and after spending a long time testing, I thought I had better check if the extension writes to PDF okay, which I did. I then transferred the PDF to my PC (form the Pi where it was generated) and printed, all good as you would expect.

I then changed the print orientation to landscape and the margins from default to minimal did again, results as expected. At that point I reset everything in the Chromium dialogue for the 'Print' extension back i.e. destination from 'PDF' back to the VNC printer, the orientation from 'landscape' back to 'portrait' and the margins from 'minimum' back to 'default'.

I am guessing that it may have been the destination switching that woke up or reset something in the printing system! Sorry for any inconvenience my original post may have caused but hopefully there is some useful info here for future use if anyone else has the same issue.
Regards,
Lee

**Comment by PeterWone** [comment]
When you use desktop remoting software like VNC or RDP, the browser used to print will be on the remote desktop because that's where VS Code is running, and this produces exactly the behaviour you describe.

The solution you came up with will certainly work but here's how I handled it when I used a Windows workstation with the code and executable on a Pi.

1. Install the [Remote SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) on your workstation.
2. Set up your Pi and your workstation for SSH using a key rather than interactive username/password.
3. Using VS Code on your workstation, connect to the remote host and open a remote workspace.

This setup will make VS Code more responsive because it has all the memory and CPU resources of your workstation. When you run your code, it will compile and run on the remote host (on the Pi). When you print, the browser will launch on your workstation and it will have access to whatever printers you can access from there.
